src/app-search/AppSearchDriver.js

```diff
@@ -0,0 +1,350 @@
+        size: resultsPerPage
+      },
+      filters: {
+        all: filters
+      }
+    };
+
+    if (sortField && sortDirection) {
+      searchOptions.sort = {
+        [sortField]: sortDirection
+      };
+    }
+
+    return this.client.search(searchTerm, searchOptions).then(resultList => {
+      this._setState({
+        current: resultList.info.meta.page.current,
+        facets: resultList.info.facets,
+        filters: filters,
+        requestId: resultList.info.meta.request_id,
+        results: resultList.results,
+        resultsPerPage: resultsPerPage,
+        searchTerm: searchTerm,
+        sortDirection: sortDirection,
+        sortField: sortField,
+        totalResults: resultList.info.meta.page.total_results
+      });
+
+      if (!skipPushToUrl) {
+        this.URLManager.pushStateToURL({
+          current,
+          filters,
+          resultsPerPage,
+          searchTerm,
+          sortDirection,
+          sortField
+        });
+      }
+    });
+  };
+
+  _setState(newState) {
+    const state = { ...this.state, ...newState };
+    this.subscriptions.forEach(subscription => subscription(state));
+    this.state = state;
+  }
+
+  /**
+   * Any time state is updated in this Driver, the provided callback
+   * will be executed with the updated state.
+   *
+   * @param onStateChange Function
+   */
+  subscribeToStateChanges(onStateChange) {
+    this.subscriptions.push(onStateChange);
+  }
+
+  /**
+   * Retrieve current state. Typically used on app initialization. Subsequent
+   * state updates should come through subscription.
+   *
+   * @returns Current state
+   */
+  getState() {
+    // We return a copy of state here, because we want to ensure the state
+    // inside of this object remains immutable.
+    return { ...this.state };
+  }
+
+  /**
+   * Filter results
+   *
+   * Will trigger new search
+   *
+   * @param name String field name to filter on
+   * @param value String field value to filter on
+   */
+  addFilter = (name, value) => {
+    const {
+      filters,
+      resultsPerPage,
+      searchTerm,
+      sortDirection,
+      sortField
+    } = this.state;
+    this._updateSearchResults({
+      current: 1,
+      filters: [...filters, { [name]: value }],
+      resultsPerPage,
+      searchTerm,
+      sortDirection,
+      sortField
+    });
+  };
+
+  /**
+   * Remove filter from results
+   *
+   * Will trigger new search
+   *
+   * @param name String field name for filter to remove
+   * @param value String field value for filter to remove
+   */
+  removeFilter = (name, value) => {
+    const {
+      filters,
+      resultsPerPage,
+      searchTerm,
+      sortDirection,
+      sortField
+    } = this.state;
+    const updatedFilters = filters.filter(filter => !(filter[name] === value));
+    this._updateSearchResults({
+      current: 1,
+      filters: updatedFilters,
+      resultsPerPage,
+      searchTerm,
+      sortDirection,
+      sortField
+    });
```

Comment:
I hadn't considered doing that. It's primarily because that method doesn't need the entire state, it just needs 5 fields from the state. It should be harmless to pass the extra fields though.

I could also possibly update the _updateSearchResults to simply do that piece of it inside of the method. So default to current state if a particular field isn't passed.